converter/src/main/java/gov/cms/qpp/conversion/validate/Checker.java

```diff
@@ -122,6 +126,42 @@ public Checker singleValue(LocalizedError code, String name) {
 		return this;
 	}
 
+	/**
+	 * checks target node for a date that is properly formatted.
+	 *
+	 * @param code that identifies the error
+	 * @param name key of the expected value
+	 * @return The checker, for chaining method calls
+	 */
+	public Checker isValidDate(LocalizedError code, String name) {
+		String date = node.getValue(name);
+		try {
+			String parse = cleanString(date);
+			parse = parse.replace("-", "").replace("/", "");
```

Comment:
I wonder if it's just easier to give the string to java.time and just see if it can make a valid datetime object out of it.  If not, then reject it.  If it's valid then we know the encoder can correctly encode it.

Reply:
This code is nearly the same as in our [encoder code](https://github.com/CMSgov/qpp-conversion-tool/blob/master/converter/src/main/java/gov/cms/qpp/conversion/encode/JsonWrapper.java#L407).  If we change this code, we should change the encoder code too.  In fact, I'd say we should make a helper class or find some appropriate existing class to hold the similar date validation logic.

Reply:
@halprin would a helper class be needed if it was changed to just try and parse the value given? I feel that it's not needed after the initial change @sfradkin suggested.

Reply:
Unfortunately, it's not as simple constructing a `LocalDateTime` object from the string. LocalDateTime uses a specific format and only allows me to specify one format to parse with. It would be limited to only one of the multiple formats we would want to accept. I realized this after starting to add additional tests.